internal/v1/handler_blueprints.go

```diff
@@ -81,30 +75,67 @@ func (bb *BlueprintBody) RedactPasswords() {
 	}
 }
 
-func (u *User) setPassword(password *string) {
-	if password != nil {
-		if u.Password == nil {
-			u.Password = new(string)
-		}
-		*u.Password = *password
+// Merges Password or SshKey from other User struct to this User struct if it is not set
+func (u *User) MergeExisting(other User) {
+	if u.Password == nil {
+		u.Password = other.Password
+	}
+	if u.SshKey == nil {
+		u.SshKey = other.SshKey
 	}
 }
 
-func (u *User) setSshKey(sshKey *string) {
-	if sshKey != nil {
-		if u.SshKey == nil {
-			u.SshKey = new(string)
-		}
-		*u.SshKey = *sshKey
+// User must have name and non-empty password or ssh key
+func (u *User) hasValidValues() error {
+	nameRe := regexp.MustCompile(`\S+`)
+	validName := nameRe.MatchString(u.Name)
+	validPassword := u.Password != nil && len(*u.Password) > 0
+	validSshKey := u.SshKey != nil && len(*u.SshKey) > 0
+	isValid := validName && (validPassword || validSshKey)
+	if !isValid {
+		return fmt.Errorf("User ('%s') must have a name and either a password or an SSH key set.", u.Name)
 	}
+	return nil
 }
 
-func (u *User) hasPassword() bool {
-	return u.Password != nil && len(*u.Password) > 0
-}
+func (u *User) isValidForUpdate(h *Handlers, ctx echo.Context, blueprintId uuid.UUID, userID *Identity) error {
```

Comment:
This method has side effect and it is very important to advertise such thing via a good method name. This name implies that it just tests for validity: `isValidForUpdate` while it actually modifies the receiver. I suggest to come up with a better name, perhaps since the main modification is called "merge" it could be also featured in the new name.